core/src/main/resources/error/error-classes.json

```diff
@@ -1877,6 +1877,11 @@
           "JOIN USING with LATERAL correlation."
         ]
       },
+      "LATERAL_WITH_PIVOT" : {
```

Comment:
The main point is the combination is not allowed in `FROM`. Since there are 3 related errors:
- unpivotWithPivotInFromClauseNotAllowedError
- lateralWithPivotInFromClauseNotAllowedError
- lateralWithUnpivotInFromClauseNotAllowedError

Could you add the error class `NOT_ALLOWED_IN_FROM`, and add sub-classes to it, please.

Reply:
Ok, Let me do it